File: 백준/Gold/2230. 수 고르기/수 고르기.cc
#include<iostream>
#include<vector>
#include<algorithm>
#include<limits.h>
using namespace std;
int n,m,ans = INT_MAX;
int arr[100005];

int main()
{
    ios::sync_with_stdio(0); cin.tie(0);    
    cin >> n >> m;
    for(int i = 0; i<n; i++) cin >> arr[i];
    sort(arr,arr+n);
    int s = 0; int e = 0;
    while(s < n && e < n){
        if(arr[e] - arr[s] >= m){
            ans = min(arr[e]-arr[s],ans);
            s++;
        }
        else e++;
    }
    cout << ans;
    return 0;
}

File: 백준/Gold/2230. 수 고르기/수 고르기.java
import java.io.*;
import java.util.*;

class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int n = Integer.parseInt(st.nextToken());
		int m = Integer.parseInt(st.nextToken());
		int[] arr = new int[n];
		for(int i = 0; i < n; i++) arr[i] = Integer.parseInt(br.readLine());
		Arrays.sort(arr);
		int ans = Integer.MAX_VALUE;
		int i = 0; int j = 0;
		while(j < n && i <= j){
			int diff = arr[j] - arr[i];
			if(diff < m) j++;
			else {
				ans = Math.min(ans, diff);
				i++;
			}
		}
		
		System.out.println(ans);
	}
}